matlab/src/cpp/arrow/matlab/array/proxy/timestamp_array.cc

```diff
@@ -49,7 +49,8 @@ namespace arrow::matlab::array::proxy {
         const mda::TypedArray<mda::MATLABString> units_mda = opts[0]["TimeUnit"];
 
         // extract the time zone string
-        MATLAB_ASSIGN_OR_ERROR(const auto timezone, arrow::util::UTF16StringToUTF8(timezone_mda[0]),
+        const std::u16string u16_timezone = timezone_mda[0];
```

Comment:
I don't know the type of `timezone_mda[0]` but does this cause a copy?

Can we use `std::u16string_view` here?

```cpp
std::u16string_view u16_timezone(timezone_mda[0]);
```

Reply:
Hi @kou, this does cause a copy. Unfortunately, it's not  possible to create a `std::816string_view` from `timezone_mda` directly. However, I can extract a `const` reference from `timezone_mda:

```cpp
const std::u16string& u16_timezone = timezone_mda[0];
```

That should avoid the copy.

Reply:
Just pushed the change.